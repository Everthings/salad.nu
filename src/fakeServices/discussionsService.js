const discussions = {
  "018269": [],
  "018308": [],
  "018298": [],
  "023188": [],
  "020307": [],
  "020310": [],
  "020311": [],
  "022682": [],
  "000968": [],
  "019985": [],
  "000998": [],
  "018037": [],
  "000971": [],
  "000985": [],
  "001010": [],
  "023596": [],
  "023755": [],
  "019652": [],
  "019828": [],
  "001007": [],
  "022323": [],
  "001006": [],
  "015579": [],
  "007031": [],
  "006942": [],
  "016014": [],
  "007000": [],
  "006915": [],
  "015327": [],
  "010968": [],
  "020843": [],
  "021390": [],
  "006982": [],
  "020757": [],
  "022037": [],
  "024054": [],
  "021300": [],
  "016304": [],
  "006991": [],
  "022768": [],
  "023399": [],
  "022705": [],
  "021724": [],
  "021426": [],
  "021237": [],
  "021581": [],
  "022952": [],
  "019158": [],
  "022512": [],
  "004969": [],
  "022525": [],
  "008177": [],
  "022536": [],
  "017947": [],
  "022139": [],
  "017948": [],
  "022140": [],
  "017949": [],
  "022141": [],
  "017950": [],
  "022174": [],
  "017951": [],
  "022175": [],
  "017952": [],
  "022176": [],
  "020079": [],
  "022142": [],
  "017953": [],
  "022143": [],
  "017954": [],
  "022144": [],
  "017955": [],
  "022145": [],
  "020081": [],
  "022147": [],
  "019009": [],
  "022150": [],
  "019010": [],
  "022151": [],
  "019011": [],
  "022152": [],
  "020082": [],
  "022153": [],
  "019196": [],
  "022154": [],
  "019197": [],
  "022155": [],
  "019198": [],
  "022156": [],
  "020080": [],
  "022157": [],
  "017956": [],
  "022158": [],
  "017957": [],
  "022159": [],
  "017958": [],
  "022160": [],
  "017959": [],
  "022177": [],
  "020076": [],
  "022161": [],
  "017960": [],
  "022162": [],
  "017961": [],
  "022163": [],
  "017962": [],
  "022164": [],
  "020077": [],
  "022165": [],
  "019012": [],
  "022166": [],
  "019013": [],
  "022167": [],
  "019014": [],
  "022168": [],
  "020083": [],
  "022169": [],
  "019816": [],
  "022170": [],
  "019814": [],
  "022171": [],
  "019815": [],
  "022172": [],
  "020078": [],
  "022173": [],
  "023292": [],
  "023293": [],
  "023294": [],
  "023295": [],
  "023296": [],
  "023297": [],
  "023298": [],
  "005455": [],
  "022298": [],
  "004994": [],
  "005010": [],
  "021367": [],
  "005121": [],
  "022539": [],
  "016320": [],
  "000804": [],
  "021571": [],
  "021816": [],
  "022029": [],
  "015585": [],
  "015574": [],
  "000850": [],
  "015392": [],
  "000822": [],
  "021927": [],
  "023646": [],
  "000830": [],
  "019840": [],
  "022412": [],
  "023693": [],
  "019843": [],
  "015695": [],
  "015500": [],
  "015577": [],
  "015484": [],
  "023028": [],
  "015700": [],
  "019048": [],
  "020131": [],
  "016508": [],
  "021101": [
    {
      unique_id: "42098-60",
      component: "LAB",
      mode: "Synchronous",
      title: "Practicum in Human Development - Washington, D.C.",
      subject: "SESP",
      number: "384-0",
      section: "60",
      meeting_days: "MoTuWeTh",
      start_time: "8:00",
      end_time: "20:00",
      room: null,
      start_date: "2021-06-21",
      end_date: "2021-08-28",
    },
  ],
  "021102": [
    {
      unique_id: "42099-60",
      component: "LAB",
      mode: null,
      title: "Practicum in Human Development - San Francisco",
      subject: "SESP",
      number: "389-0",
      section: "60",
      meeting_days: "MoTuWeTh",
      start_time: "8:00",
      end_time: "20:00",
      room: null,
      start_date: "2021-06-21",
      end_date: "2021-08-28",
    },
  ],
  "023918": [
    {
      unique_id: "42100-60",
      component: "LAB",
      mode: "Synchronous",
      title: "Experiential Learning: Practicum",
      subject: "SESP",
      number: "392-0",
      section: "60",
      meeting_days: "MoTuWeTh",
      start_time: "8:00",
      end_time: "20:00",
      room: null,
      start_date: "2021-06-21",
      end_date: "2021-08-28",
    },
    {
      unique_id: "42101-61",
      component: "LAB",
      mode: "Synchronous",
      title: "Experiential Learning: Practicum",
      subject: "SESP",
      number: "392-0",
      section: "61",
      meeting_days: "TuWeThFr",
      start_time: "8:00",
      end_time: "20:00",
      room: null,
      start_date: "2021-06-21",
      end_date: "2021-08-28",
    },
    {
      unique_id: "42102-62",
      component: "LAB",
      mode: "Synchronous",
      title: "Experiential Learning: Practicum",
      subject: "SESP",
      number: "392-0",
      section: "62",
      meeting_days: "MoTuWeTh",
      start_time: "8:00",
      end_time: "20:00",
      room: null,
      start_date: "2021-06-21",
      end_date: "2021-08-28",
    },
  ],
  "017367": [],
  "022279": [],
  "019649": [],
  "004368": [],
  "004134": [],
  "004192": [],
  "021922": [],
  "004206": [],
  "004242": [],
  "020248": [],
  "022882": [],
  "022886": [],
  "011480": [],
  "022888": [],
  "004451": [],
  "021070": [],
  "004477": [],
  "021716": [],
  "021719": [],
  "021718": [],
  "004559": [],
  "022876": [],
  "004464": [],
  "023687": [],
  "022371": [],
  "022374": [],
  "015981": [],
  "023690": [],
  "023688": [],
  "015985": [],
  "021261": [],
  "021263": [],
  "021264": [],
  "015460": [],
  "022106": [],
  "004308": [],
  "022350": [],
  "022095": [],
  "019880": [],
  "022237": [],
  "022400": [],
  "021648": [],
  "022250": [],
  "021233": [],
  "023242": [],
  "021651": [],
  "020372": [],
  "004171": [],
  "004336": [],
  "015210": [],
  "004343": [],
  "004373": [],
  "011778": [
    {
      unique_id: "41593-65",
      component: "LAB",
      mode: "Synchronous",
      title: "Topics in Interactive Media",
      subject: "RTVF",
      number: "376-0",
      section: "65",
      meeting_days: "Fr",
      start_time: "10:00",
      end_time: "11:50",
      room: null,
      start_date: "2021-06-21",
      end_date: "2021-07-25",
    },
  ],
  "021674": [],
  "022181": [],
  "022474": [],
  "020325": [],
  "020798": [],
  "022935": [],
  "022933": [],
  "022900": [],
  "023672": [],
  "020458": [],
  "020459": [],
  "020460": [],
  "020461": [],
  "020467": [],
  "021905": [],
  "021098": [],
  "016362": [],
  "022047": [],
  "018466": [],
  "023615": [],
  "021384": [],
  "021380": [],
  "016549": [],
  "024019": [],
  "017771": [],
  "017526": [],
  "021007": [],
  "022842": [],
  "019168": [],
  "021575": [],
  "023377": [],
  "023439": [],
  "022238": [],
  "022302": [],
  "023430": [],
  "020808": [],
  "020810": [],
  "020966": [],
  "022930": [],
  "020967": [],
  "023331": [],
  "022761": [],
  "022762": [],
  "017630": [],
  "021465": [],
  "006181": [],
  "023376": [],
  "023664": [],
  "023665": [],
  "018840": [],
  "017707": [],
  "017709": [],
  "023760": [],
  "013506": [],
  "005925": [],
  "017664": [],
  "020333": [],
  "023404": [],
  "016182": [],
  "016183": [],
  "017669": [],
  "021838": [],
  "018092": [],
  "017853": [],
  "021839": [],
  "022453": [],
  "022309": [],
  "018585": [],
  "018591": [],
  "018592": [],
  "018593": [],
  "018594": [],
  "016585": [],
  "022892": [],
  "023650": [],
  "022254": [],
  "022253": [],
  "020971": [],
  "020631": [],
  "020741": [],
  "018600": [],
  "022271": [],
  "022835": [],
  "022387": [],
  "022458": [],
  "022386": [],
  "022009": [],
  "024040": [],
  "021842": [],
  "017881": [],
  "019179": [],
  "022776": [],
  "022777": [],
  "022778": [],
  "023637": [],
  "022779": [],
  "022780": [],
  "022781": [],
  "022782": [],
  "022783": [],
  "022784": [],
  "022785": [],
  "022786": [],
  "023493": [],
  "022790": [],
  "022791": [],
  "022793": [],
  "022794": [],
  "022796": [],
  "022798": [],
  "023364": [],
  "022799": [],
  "022801": [],
  "022802": [],
  "022803": [],
  "023912": [],
  "022808": [],
  "021129": [],
  "021130": [],
  "021122": [],
  "023862": [],
  "021433": [],
  "023431": [],
  "023592": [],
  "023593": [],
  "023594": [],
  "023715": [],
  "023378": [],
  "022092": [],
  "023821": [],
  "023829": [],
  "021532": [],
  "021533": [],
  "021534": [],
  "021536": [],
  "021537": [],
  "021538": [],
  "021539": [],
  "021540": [],
  "021545": [],
  "021547": [],
  "021549": [],
  "023924": [],
  "017670": [],
  "017671": [],
  "017672": [],
  "023521": [],
  "018337": [],
  "005864": [],
  "022452": [],
  "024022": [],
  "022451": [],
  "020921": [],
  "023869": [],
  "018238": [],
  "020484": [],
  "021523": [],
  "023424": [],
  "021500": [],
  "020382": [],
  "023652": [],
  "020383": [],
  "022015": [],
  "021327": [],
  "022286": [],
  "023475": [],
  "023401": [],
  "019257": [],
  "023705": [],
  "008702": [],
  "001195": [],
  "016191": [],
  "001218": [],
  "001094": [],
  "001095": [],
  "001096": [],
  "001097": [],
  "015103": [],
  "001147": [],
  "002324": [],
  "002325": [],
  "002326": [],
  "002327": [],
  "002328": [],
  "002329": [],
  "001309": [],
  "001335": [],
  "001339": [],
  "019819": [],
  "019886": [],
  "019885": [],
  "019888": [],
  "019889": [],
  "019890": [],
  "001389": [],
  "018633": [],
  "021511": [],
  "020519": [],
  "017400": [],
  "017401": [],
  "017402": [],
  "017403": [],
  "017782": [],
  "017783": [],
  "018515": [],
  "020001": [],
  "019546": [],
  "019818": [],
  "019951": [],
  "022460": [],
  "022461": [],
  "022462": [],
  "022463": [],
  "022464": [],
  "021735": [],
  "001467": [],
  "001468": [],
  "001469": [],
  "020687": [],
  "020688": [],
  "002331": [],
  "002332": [],
  "002333": [],
  "002334": [],
  "002335": [],
  "002336": [],
  "001605": [],
  "023045": [],
  "018661": [],
  "021746": [],
  "001646": [],
  "001647": [],
  "001655": [],
  "001657": [],
  "001658": [],
  "001659": [],
  "001673": [],
  "001679": [],
  "001680": [],
  "022402": [],
  "019446": [],
  "022061": [],
  "001767": [],
  "022206": [],
  "001791": [],
  "001792": [],
  "001797": [],
  "001844": [],
  "001845": [],
  "023333": [],
  "018572": [],
  "020567": [
    {
      unique_id: "41993-21",
      component: "LAB",
      mode: "Asynchronous",
      title: "Special Topics in Environmental Sciences",
      subject: "ENVR_SCI",
      number: "390-0",
      section: "21",
      meeting_days: "MoWeFr",
      start_time: "15:00",
      end_time: "16:20",
      room: null,
      start_date: "2021-06-21",
      end_date: "2021-07-18",
    },
  ],
  "002987": [],
  "002988": [],
  "002989": [],
  "002993": [],
  "002994": [],
  "002995": [],
  "021495": [],
  "016805": [],
  "001955": [],
  "001956": [],
  "001957": [],
  "017811": [],
  "017817": [],
  "003425": [],
  "018651": [],
  "002030": [],
  "002031": [],
  "022479": [],
  "023920": [],
  "003078": [],
  "003079": [],
  "003080": [],
  "003094": [],
  "002349": [],
  "002350": [],
  "002351": [],
  "018748": [],
  "018649": [],
  "020156": [],
  "002391": [],
  "002390": [],
  "002417": [],
  "002436": [],
  "021793": [],
  "021792": [],
  "021747": [],
  "023340": [],
  "023341": [],
  "023345": [],
  "023346": [],
  "008660": [],
  "002472": [],
  "002492": [],
  "002596": [],
  "002602": [],
  "019742": [],
  "019741": [],
  "002609": [],
  "008550": [],
  "002613": [],
  "016061": [],
  "015860": [],
  "002703": [],
  "002704": [],
  "002705": [],
  "002706": [],
  "002707": [],
  "002708": [],
  "021527": [],
  "021528": [],
  "021529": [],
  "002768": [],
  "017872": [],
  "020037": [],
  "020515": [],
  "021554": [],
  "002818": [],
  "002820": [],
  "003127": [],
  "003128": [],
  "003129": [],
  "002874": [],
  "002875": [],
  "002876": [],
  "002879": [],
  "002881": [],
  "002884": [],
  "022516": [],
  "002895": [],
  "019120": [],
  "002218": [],
  "002226": [],
  "002257": [],
  "003306": [],
  "003312": [],
  "003316": [],
  "019402": [],
  "003318": [],
  "003136": [],
  "003137": [],
  "003138": [],
  "003139": [],
  "003140": [],
  "003141": [],
  "015856": [],
  "003157": [],
  "003158": [],
  "003388": [],
};

export function getDiscussion(courseId) {
  return discussions[courseId] || [];
}
