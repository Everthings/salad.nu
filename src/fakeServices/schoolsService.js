const schools = [
  { symbol: "DOHA", name: "Northwestern in Qatar" },
  { symbol: "JOUR", name: "Medill School of Journalism" },
  { symbol: "KGSM", name: "J L Kellogg  School Management" },
  { symbol: "LAW", name: "Law School" },
  {
    symbol: "MEAS",
    name: "McCormick School of Engineering and Applied Science",
  },
  { symbol: "MUSIC", name: "Bienen School of Music" },
  { symbol: "NDGR", name: "Non-Degree Schools" },
  { symbol: "SESP", name: "School of Educ & Social Policy" },
  { symbol: "SPCH", name: "School of Communication" },
  { symbol: "TGS", name: "The Graduate School" },
  { symbol: "UC", name: "School of Professional Studies" },
  { symbol: "WCAS", name: "Weinberg College of Arts and Sciences" },
];

export function getSchools() {
  return schools.filter((school) => school);
}
