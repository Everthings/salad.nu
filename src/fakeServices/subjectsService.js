const subjects = {
  JOUR: [
    { symbol: "IMC", name: "Integrated Marketing Communications" },
    { symbol: "JOUR", name: "Journalism" },
  ],
  LAW: [
    { symbol: "BUSCOM", name: "Business and Commercial Law" },
    { symbol: "CONPUB", name: "Constitutional and Public Law" },
    { symbol: "CRIM", name: "Criminal Law" },
    { symbol: "INTPROP", name: "Intellectual Property" },
    { symbol: "LAWSTUDY", name: "Law Studies-General or Interdisciplinary" },
    { symbol: "LITARB", name: "Litigation and Arbitration" },
    { symbol: "PPTYTORT", name: "Property and Tort Law" },
    { symbol: "REGLAW", name: "Regulatory Law" },
    { symbol: "TAXLAW", name: "Taxation Law" },
  ],
  MEAS: [
    { symbol: "BMD_ENG", name: "Biomedical Engineering" },
    { symbol: "CHEM_ENG", name: "Chemical Engineering" },
    { symbol: "CIV_ENV", name: "Civil and Environmental Engineering" },
    { symbol: "COMP_ENG", name: "Computer Engineering" },
    { symbol: "COMP_SCI", name: "Computer Science" },
    { symbol: "CRDV", name: "Career Development" },
    { symbol: "DSGN", name: "Segal Design Institute" },
    { symbol: "ELEC_ENG", name: "Electrical Engineering" },
    { symbol: "ENTREP", name: "Entrepreneurship & Innovation" },
    { symbol: "ES_APPM", name: "Engineering Science & Applied Math" },
    { symbol: "GEN_ENG", name: "General Engineering" },
    { symbol: "IEMS", name: "Industrial Engineering & Management Sciences" },
    { symbol: "INF_TECH", name: "Information Technology" },
    { symbol: "ISEN", name: "Initiative for Sustainability & Energy at NU" },
    { symbol: "LDRSHP", name: "Center for Leadership" },
    { symbol: "MAT_SCI", name: "Materials Science & Engineering" },
    { symbol: "MBIOTECH", name: "Masters in Biotechnology Program" },
    { symbol: "MECH_ENG", name: "Mechanical Engineering" },
    { symbol: "MEM", name: "Master of Engineering Management" },
    { symbol: "MSAI", name: "Artificial Intelligence" },
    { symbol: "MSIA", name: "Master of Science in Analytics" },
    { symbol: "PRDV", name: "Personal Development" },
    { symbol: "PROJ_MGT", name: "Project Management" },
    { symbol: "PSED", name: "Predictive Science and Engineering" },
  ],
  SESP: [
    { symbol: "HDC", name: "Human Development in Context" },
    { symbol: "HDSP", name: "Human Development & Social Policy" },
    { symbol: "LOC", name: "Learning & Organizational Change" },
    { symbol: "LRN_SCI", name: "Learning Sciences" },
    { symbol: "MS_ED", name: "MS in Education & Social Policy" },
    {
      symbol: "MS_HE",
      name: "MS in Higher Education Administration and Policy",
    },
    { symbol: "MS_LOC", name: "Learning & Organizational Change" },
    { symbol: "SESP", name: "School of Education & Social Policy Core" },
    { symbol: "SE_POL", name: "Social & Economic Policy" },
    { symbol: "SOC_POL", name: "Social Policy" },
    { symbol: "TEACH_ED", name: "Teacher Education" },
  ],
  SPCH: [
    { symbol: "CMN", name: "Communication Related Courses" },
    { symbol: "COMM_ST", name: "Communication Studies" },
    { symbol: "CSD", name: "Communication Sciences and Disorders" },
    { symbol: "DANCE", name: "Dance" },
    { symbol: "HLTH_COM", name: "Health Communication" },
    { symbol: "MSC", name: "Master of Science in Communication" },
    { symbol: "MSLCE", name: "MS Leadership for Creative Enterprises" },
    { symbol: "MTS", name: "Media, Technology & Society" },
    { symbol: "PERF_ST", name: "Performance Studies" },
    { symbol: "RTVF", name: "Radio/Television/Film" },
    { symbol: "SAI", name: "Sound Arts and Industries" },
    { symbol: "THEATRE", name: "Theatre" },
  ],
  TGS: [
    { symbol: "APP_PHYS", name: "Applied Physics" },
    { symbol: "BIOSTAT", name: "Biostatistics" },
    { symbol: "CHSS", name: "Comparative & Historical Social Science" },
    { symbol: "CIC", name: "CIC Traveling Scholar" },
    { symbol: "CLIN_PSY", name: "Clinical Psychology" },
    { symbol: "CME", name: "Chicago Metropolitan Exchange Program" },
    { symbol: "COUN", name: "Counseling" },
    { symbol: "DATA_SCI", name: "Data Science" },
    { symbol: "DGP", name: "Driskill Graduate Program in Life Sciences" },
    { symbol: "GENET_CN", name: "Genetic Counseling" },
    { symbol: "HBMI", name: "Health and Biomedical Informatics" },
    { symbol: "HQS", name: "Healthcare Quality and Patient  Safety" },
    { symbol: "HSIP", name: "Health Sciences Integrated Program" },
    { symbol: "HSR", name: "Health Services and Outcomes Research" },
    { symbol: "LRN_SCI", name: "Learning Sciences" },
    { symbol: "MENA", name: "Middle East and North African Studies" },
    { symbol: "MHB", name: "Medical Humanities and Bioethics" },
    { symbol: "MSCI", name: "Master of Science in Clinical Investigation" },
    { symbol: "MS_FT", name: "MS in Marital & Family Therapy" },
    { symbol: "NUIN", name: "Neuroscience" },
    { symbol: "PBC", name: "Plant Biology & Conservation" },
    { symbol: "PUB_HLTH", name: "Public Health" },
    { symbol: "QSB", name: "Quantitative and Systems Biology" },
    { symbol: "REPR_SCI", name: "Reproductive Science and Medicine" },
    { symbol: "TGS", name: "TGS General Registrations" },
    { symbol: "THEATRE", name: "Theatre" },
  ],
  WCAS: [
    { symbol: "AFST", name: "African Studies" },
    { symbol: "AF_AM_ST", name: "African American Studies" },
    { symbol: "AMER_ST", name: "American Studies" },
    { symbol: "ANTHRO", name: "Anthropology" },
    { symbol: "ARABIC", name: "Arabic" },
    { symbol: "ART", name: "Art Theory & Practice" },
    { symbol: "ART_HIST", name: "Art History" },
    { symbol: "ASIAN_AM", name: "Asian American Studies" },
    { symbol: "ASIAN_LC", name: "Asian Languages and Cultures" },
    { symbol: "ASTRON", name: "Astronomy" },
    { symbol: "BIOL_SCI", name: "Biological Sciences" },
    { symbol: "BUS_INST", name: "Business Institutions" },
    { symbol: "CFS", name: "Chicago Field Studies" },
    { symbol: "CHEM", name: "Chemistry" },
    { symbol: "CHINESE", name: "Chinese" },
    { symbol: "CLASSICS", name: "Classics - Readings in English" },
    { symbol: "COG_SCI", name: "Cognitive Science" },
    { symbol: "COMP_LIT", name: "Comparative Literary Studies" },
    { symbol: "EARTH", name: "Earth and Planetary Sciences" },
    { symbol: "ECON", name: "Economics" },
    { symbol: "ENGLISH", name: "English" },
    { symbol: "ENVR_POL", name: "Environmental Policy & Culture" },
    { symbol: "ENVR_SCI", name: "Environmental Science" },
    { symbol: "FRENCH", name: "French" },
    { symbol: "GAMS", name: "Global Avant-garde and Modernist Studies" },
    { symbol: "GBL_HLTH", name: "Global Health" },
    { symbol: "GEN_LA", name: "General Liberal Arts" },
    { symbol: "GEOG", name: "Geography" },
    { symbol: "GERMAN", name: "German" },
    { symbol: "GNDR_ST", name: "Gender Studies" },
    { symbol: "GREEK", name: "Greek" },
    { symbol: "HEBREW", name: "Hebrew" },
    { symbol: "HIND_URD", name: "Hindi and Urdu" },
    { symbol: "HISTORY", name: "History" },
    { symbol: "HUM", name: "Humanities" },
    { symbol: "IBIS", name: "Interdisciplinary Biological Sciences Program" },
    { symbol: "INTG_SCI", name: "Integrated Science" },
    { symbol: "INTL_ST", name: "International Studies" },
    { symbol: "ISEN", name: "Initiative for Sustainability & Energy at NU" },
    { symbol: "ITALIAN", name: "Italian" },
    { symbol: "JAPANESE", name: "Japanese" },
    { symbol: "JWSH_ST", name: "Jewish Studies" },
    { symbol: "KOREAN", name: "Korean" },
    { symbol: "LATIN", name: "Latin" },
    { symbol: "LATINO", name: "Latina and Latino Studies" },
    { symbol: "LEGAL_ST", name: "Legal Studies" },
    { symbol: "LING", name: "Linguistics" },
    { symbol: "MATH", name: "Mathematics" },
    { symbol: "MENA", name: "Middle East and North African Studies" },
    { symbol: "MMSS", name: "Math Methods in the Social Sciences" },
    { symbol: "NEUROBIO", name: "Neurobiology & Physiology" },
    { symbol: "NEUROSCI", name: "Neuroscience" },
    { symbol: "PERSIAN", name: "Persian" },
    { symbol: "PHIL", name: "Philosophy" },
    { symbol: "PHYSICS", name: "Physics" },
    { symbol: "POLISH", name: "Polish" },
    { symbol: "POLI_SCI", name: "Political Science" },
    { symbol: "PORT", name: "Portuguese" },
    { symbol: "PSYCH", name: "Psychology" },
    { symbol: "RELIGION", name: "Religious Studies" },
    { symbol: "RUSSIAN", name: "Russian" },
    { symbol: "SLAVIC", name: "Slavic Languages & Literature" },
    { symbol: "SOCIOL", name: "Sociology" },
    { symbol: "SPANISH", name: "Spanish" },
    { symbol: "SPANPORT", name: "Spanish and Portuguese" },
    { symbol: "STAT", name: "Statistics" },
    { symbol: "SWAHILI", name: "Swahili" },
    { symbol: "TRANS", name: "Transportation" },
    { symbol: "TURKISH", name: "Turkish" },
  ],
};

export function getSubject(school) {
  return subjects[school] || [];
}
