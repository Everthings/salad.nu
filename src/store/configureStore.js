import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
import rootReducer from "./reducer";
import api from "./middleware/api";

export default function () {
  return configureStore({
    reducer: rootReducer,
    middleware: [...getDefaultMiddleware(), api],
  });
}
