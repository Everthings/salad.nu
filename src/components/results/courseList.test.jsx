import React from "react";
import ReactDOM from "react-dom";
import { render, waitForElement } from "@testing-library/react";
import { ToastProvider } from "react-toast-notifications";
import configureStore from "./../../store/configureStore";
import { Provider } from "react-redux";
import * as coursesService from "./../../fakeServices/coursesService";
import Theme from "./../themes/theme";
import CourseList from "./courseList";

jest.mock("./../../fakeServices/coursesService");
coursesService.getCourses.mockImplementation(({ searchStr }) => {
  if (searchStr === "empty list") {
    return [];
  }

  if (searchStr === "search") {
    return [
      {
        unique_id: 1,
        title: "Title 1",
        subject: "Subject 1",
        number: "Number 1",
      },
      {
        unique_id: 2,
        title: "Title 2",
        subject: "Subject 2",
        number: "Number 2",
      },
      {
        unique_id: 3,
        title: "Title 3",
        subject: "Subject 3",
        number: "Number 3",
      },
    ];
  }
});

ReactDOM.createPortal = (node) => node;

describe("CourseList", () => {
  it("renders course list correctly with data", async () => {
    const state = {
      search: {
        searchStr: "search",
      },
      interactions: {
        selectedCourse: { id: 1 },
      },
    };
    const store = configureStore(state);
    const { getByTestId, asFragment } = render(
      <ToastProvider>
        <Provider store={store}>
          <Theme>
            <CourseList />
          </Theme>
        </Provider>
      </ToastProvider>
    );

    await waitForElement(() => getByTestId("course-list"));

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders 'No Results' if sections list is empty and search string is not empty", async () => {
    const state = {
      search: {
        searchStr: "empty list",
      },
      interactions: {
        selectedCourse: { id: -1 },
      },
    };
    const store = configureStore(state);
    const { queryByText } = render(
      <ToastProvider>
        <Provider store={store}>
          <Theme>
            <CourseList />
          </Theme>
        </Provider>
      </ToastProvider>
    );

    const noResultsText = await waitForElement(() => queryByText("No Results"));

    expect(noResultsText).not.toBeNull();
  });

  it("renders 'Continue typing...' if sections list is empty and search string is not empty", async () => {
    const state = {
      search: { searchStr: "t h i s i s t o o s h o r t" },
    };
    const store = configureStore(state);
    const { queryByText } = render(
      <ToastProvider>
        <Provider store={store}>
          <Theme>
            <CourseList />
          </Theme>
        </Provider>
      </ToastProvider>
    );

    const continueTypingText = await waitForElement(() =>
      queryByText("Continue typing...")
    );

    expect(continueTypingText).not.toBeNull();
  });
});
