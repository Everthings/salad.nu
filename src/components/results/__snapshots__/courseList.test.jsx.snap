// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseList renders course list correctly with data 1`] = `
Array [
  <div
    className="sc-Axmtr gGWzet"
    data-testid="course-list"
  >
    <div
      className="sc-AxheI eBXIT"
    >
      3
       result(s)
    </div>
    <div
      className="sc-AxjAm cFVyGa"
      data-testid="info-card"
      onClick={[Function]}
    >
      <div
        className="card-body"
      >
        <h5
          className="sc-AxiKw iiwGCz"
          data-testid="info-card-title"
        >
          Subject 1 Number 1
        </h5>
        <hr
          className="sc-AxirZ grXYcG"
        />
        <p
          className="sc-AxhCb kqIPop"
        >
          Title 1
        </p>
      </div>
    </div>
    <div
      className="sc-AxjAm cFVyGa"
      data-testid="info-card"
      onClick={[Function]}
    >
      <div
        className="card-body"
      >
        <h5
          className="sc-AxiKw iiwGCz"
          data-testid="info-card-title"
        >
          Subject 2 Number 2
        </h5>
        <hr
          className="sc-AxirZ grXYcG"
        />
        <p
          className="sc-AxhCb kqIPop"
        >
          Title 2
        </p>
      </div>
    </div>
    <div
      className="sc-AxjAm cFVyGa"
      data-testid="info-card"
      onClick={[Function]}
    >
      <div
        className="card-body"
      >
        <h5
          className="sc-AxiKw iiwGCz"
          data-testid="info-card-title"
        >
          Subject 3 Number 3
        </h5>
        <hr
          className="sc-AxirZ grXYcG"
        />
        <p
          className="sc-AxhCb kqIPop"
        >
          Title 3
        </p>
      </div>
    </div>
  </div>,
  <div
    className="react-toast-notifications__container css-mrh2d-ToastContainer"
  />,
]
`;
